doc_indexer/src/fetcher/fetcher.py

```diff
@@ -0,0 +1,70 @@
+import os
+import shutil
+
+from fetcher.scroller import Scroller
+from fetcher.source import DocumentsSource, get_documents_sources
+
+from utils.settings import LOG_LEVEL
+from utils.utils import clone_repo, get_logger
+
+logger = get_logger(__name__, LOG_LEVEL)
+
+
+class DocumentsFetcher:
+    """Class to fetch the documents from the specified sources."""
+
+    output_dir: str
+    tmp_dir: str
+    sources: list[DocumentsSource]
+
+    def __init__(self, source_file: str, output_dir: str, tmp_dir: str) -> None:
+        """Initializes the DocumentsFetcher class."""
+        self.output_dir = output_dir
+        self.tmp_dir = tmp_dir
+
+        # delete the directories if they exist
```

Comment:
I have set flags `ignore_errors=True` and `exist_ok=True`. If it fails due to any other reason then it should raise error and the script should stop so that we can manually verify the issue.